lib/benchpark/experiment.py

```diff
@@ -127,16 +137,21 @@ class Experiment(ExperimentSystemBase, SingleNode):
         description="Append to environment PATH during experiment execution",
     )
 
+    @requires_method("setup_default_experiment")
     def __init__(self, spec):
         self.spec: "benchpark.spec.ConcreteExperimentSpec" = spec
         super().__init__()
         self.helpers = []
         self._spack_name = None
         self._ramble_name = None
+        self._expr_vars = VariableDict()
+        self._required_vars = []
 
+        visited_helper_impls = set()
         for cls in self.__class__.mro()[1:]:
             if cls is not Experiment and cls is not object:
-                if hasattr(cls, "Helper"):
+                if hasattr(cls, "Helper") and cls.Helper not in visited_helper_impls:
+                    visited_helper_impls.add(cls.Helper)
```

Comment:
@scheibelp Changed this to ensure the same Helper is not called twice if extended by a child class